### PR TITLE
Feat/#tc 28 create order details page

### PR DESCRIPTION
**Pull request by CristiPV**
### This pull request closes:

✔️ Closes: #
🏁 Jira Issue: [TC-]()

### Which service(s) does this issue affect:

- [ ] 🛰️ API
- [ ] 🛒 webstore
- [x] 🎓 admin-page

### Types of changes

- [ ] 🐛 Bug fix (a non-breaking change which fixes an issue)
- [x] 🧱 New feature (a non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🤖 CI/CD (a change to the CI/CD pipeline)
- [ ] ⚗️ Refactoring (an update to some already existing code)

### Documentation Updates

- [x] 📂 No need for updates
- [ ] 📁 Requires an update ( _not done within this issue_ )
- [ ] 🗃️ Has been updated ( _done within this issue_ )

### Testing checklist

- [x] 💪 Manual tests
- [x] 🔧 Automatic tests

#### Browser compatibility - _**[**Frontend only**]**_

- [ ] Tested on Safari
- [x] Tested on Chrome
- [x] Tested on Mozilla Firefox

### What changes have been done within this issue?

I have created the orders details page.

### How should this be manually tested?

From the orders page, click the View button beside any of the orders.
It should redirect to that order's details page.

### CODE REVIEW

File: apps/admin-page/src/app/services/orders/orders.service.ts [hunk 1]
```diff
@@ -28,4 +28,83 @@ export class OrdersService {
...
+        return LabelColorsEnum.blue;
+      case OrderStatusEnum.delivered:
+        return LabelColorsEnum.green;
+      default:
+        return LabelColorsEnum.lightGrey;
+    }
+  }
+}
+
+export enum LabelColorsEnum {
```
Comment (Kwandes) [inline apps/admin-page/src/app/services/orders/orders.service.ts:103]: The enum should be located in the `@interfaces` library like the rest of enums unless it is specific to only this component (in which case there is no point exporting it)

File: apps/admin-page/src/app/pages/order-details/order-details.component.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        this.addressOneControl = new FormControl(this.order?.contactInfo.streetAddress);
+        this.addressTwoControl = new FormControl(this.order?.contactInfo.streetAddress2);
+        // Billing Information
+        this.billingNameControl = new FormControl(this.order?.billingInfo.name);
+        this.billingCityControl = new FormControl(this.order?.billingInfo.city);
+        this.billingPostcodeControl = new FormControl(this.order?.billingInfo.postcode);
+        this.billingAddressOneControl = new FormControl(this.order?.billingInfo.streetAddress);
+        this.billingAddressTwoControl = new FormControl(this.order?.billingInfo.streetAddress2);
+
+        if (this.order?.transactionItems) {
```
Comment (Teodor25) [inline apps/admin-page/src/app/pages/order-details/order-details.component.ts:130]: An order cant Not have at least one transaction item, so this if statement is not needed.

**Review by Teodor25 — APPROVED** [review]

File: apps/admin-page/src/app/pages/order-details/order-details.component.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      error: (error: HttpErrorResponse) => {
+        console.error(error);
+      },
+      next: (orders: IOrder[]) => {
+        this.isLoading = false;
+        this.order = orders.find((order) => order.orderId === id);
+        // Setting the page title.
+        this.title = `Order by: ${this.order?.contactInfo.name}`;
+
+        // Customer Information
```
Comment (Teodor25) [inline apps/admin-page/src/app/pages/order-details/order-details.component.ts:108]: There are no validators, i think we should add them.

File: apps/admin-page/src/app/pages/order-details/order-details.component.html [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+                <div class="row">
+                    <mat-form-field appearance="fill">
+                        <mat-label>Address</mat-label>
+                        <input matInput type="text" [formControl]="addressTwoControl" />
+                    </mat-form-field>
+                </div>
+            </mat-grid-tile>
+
+            <!-- Separator - Billing address -->
+            <mat-grid-tile colspan="2" rowspan="2" data-cy="billing-info-divider">
```
Comment (Teodor25) [inline apps/admin-page/src/app/pages/order-details/order-details.component.html:294]: None of the billing info fields can be updated, so the inputfields should be disabled.

File: apps/admin-page/src/app/pages/order-details/order-details.component.html [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+                        <mat-label>City</mat-label>
+                        <input matInput type="text" [formControl]="billingCityControl" />
+                    </mat-form-field>
+                    <div class="spacer"></div>
+                    <mat-form-field appearance="fill">
+                        <mat-label>Postcode</mat-label>
+                        <input matInput type="number" [formControl]="billingPostcodeControl" />
+                    </mat-form-field>
+                </div>
+                <div class="row">
```
Comment (Teodor25) [inline apps/admin-page/src/app/pages/order-details/order-details.component.html]: Should not hav e an update billing address. This cant be changed at any time for legal reasons.

**Review by Kwandes — APPROVED** [review]

**Pull request merged**